addon/components/o-s-s/phone-number-input.ts

```diff
@@ -45,14 +51,33 @@ export default class OSSPhoneNumberInput extends Component<OSSPhoneNumberInputAr
     }
   }
 
+  private validateInput(): void {
+    this.invalidInputError = '';
+
+    if (this.args.number.startsWith('+')) {
+      this.invalidInputError = this.intl.t('oss-components.phone-input.invalid_input');
+      this.args.validates?.(false);
+    } else {
+      this.args.validates?.(true);
+    }
+  }
+
   @action
-  onlyNumeric(event: KeyboardEvent): void {
-    const authorizedInputs = ['Backspace', 'Delete', 'ArrowLeft', 'ArrowRight', 'Tab', 'Shift', 'Control'];
-    if (/^[0-9]$/i.test(event.key) || authorizedInputs.find((key: string) => key === event.key)) {
+  onlyNumeric(event: KeyboardEvent | FocusEvent): void {
+    const authorizedInputs = ['Backspace', 'Delete', 'ArrowLeft', 'ArrowRight', 'Tab', 'Shift', 'Control', 'Keydown'];
+
+    if (
+      event.type === 'keydown' ||
+      event instanceof FocusEvent ||
+      /^[0-9]$/i.test(event.key) ||
+      authorizedInputs.find((key: string) => key === event.key)
+    ) {
```

Comment:
You sure about this ?
Seems to me that every keydown event will trigger this onChange method